custom/icds_reports/utils/aggregation_helpers/daily_feeding_forms_child_health.py

```diff
@@ -25,21 +25,22 @@ def aggregation_query(self):
 
         return """
         INSERT INTO "{tablename}" (
-          state_id, month, case_id, latest_time_end_processed, sum_attended_child_ids,
-          lunch_count
+          state_id, supervisor_id, month, case_id, latest_time_end_processed,
+          sum_attended_child_ids, lunch_count
         ) (
           SELECT
             %(state_id)s AS state_id,
+            LAST_VALUE(supervisor_id) OVER w AS supervisor_id,
             %(month)s AS month,
-            child_health_case_id AS case_id,
+            DISTINCT child_health_case_id AS case_id,
             MAX(timeend) AS latest_time_end_processed,
             SUM(attended_child_ids) AS sum_attended_child_ids,
             SUM(lunch) AS lunch_count
           FROM "{ucr_tablename}"
           WHERE state_id = %(state_id)s AND
                 timeend >= %(current_month_start)s AND timeend < %(next_month_start)s AND
                 child_health_case_id IS NOT NULL
-          GROUP BY child_health_case_id
+          WINDOW w AS (PARTITION BY child_health_case_id)
```

Comment:
@snopoke Does this look, correct?

Reply:
I think so, I did a test of similar query structure (not same) on a different table.

Are there any test suites for these queries?

Reply:
See comment above about changing columns to use `OVER`. There should be tests and I think these changes will make them fail since you'll need to add supervisor_id columns with data into the test CSV files: https://github.com/dimagi/commcare-hq/blob/1ee5097c223fc575bad48b6427f37786d5e1358f/custom/icds_reports/tests/fixtures/dashboard_daily_feeding.csv#L21

You'll want to use [Q](https://harelba.github.io/q/) to update the CSV files. EG:
```
q "SELECT df.*, c.supervisor_id FROM dashboard_daily_feeding.csv df JOIN child_cases.csv c ON (df.child_health_case_id = c.case_id)" -d, -H -O
```

Reply:
This only tests the UCR spec for correctness, right? I haven't been able to find tests that would check the result of aggregation queries that work on top of this UCR data. Are there any?

Reply:
all the tests in `custom/icds_reports/tests/test_aggregation_script.py`

Reply:
Yeah, that doesn't seem to have tests that test the validity of the queries that I am running and the result of those queries.

Reply:
right, you are correct that there is no test validating the output of the query. Shouldn't be hard to make one though which I think we should do for any query we are changing in more ways than just adding the supervisor column.

Reply:
So, I offlined with @calellowitz a bit on this . It looks like since the existing agg tests are integration tests, if they succeed, it would mean that these intermediate dashboard tables got built correctly, in which we don't need to write tests for validating these dashboard tables (unless we specifically want to test correct  `supervisor_id` is added).

Reply:
ok that seem ok